File: src/com/fate/engine/math/VectorUtils.java
package com.fate.engine.math;

public class VectorUtils {
	public static float triangleArea(Vector2f v1, Vector2f v2, Vector2f v3) {
		Vector2f p1 = v1.sub(v3);
		Vector2f p2 = v2.sub(v3);
		
		return Math.abs(p1.getX() * p2.getY() - p1.getY() * p2.getX()) / 2;
	}
}

File: src/main/kotlin/com/morph/engine/math/VectorUtils.kt
package com.morph.engine.math

object VectorUtils {
    fun triangleArea(v1: Vector2f, v2: Vector2f, v3: Vector2f): Float {
        val p1 = v1.sub(v3)
        val p2 = v2.sub(v3)

        return Math.abs(p1.x * p2.y - p1.y * p2.x) / 2
    }
}
